File: sort/selection.js
function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIdx = i;
    for (j = i + 1; j < arr.length; j++) {
      if (arr[minIdx] > arr[j]) {
        minIdx = j;
      }
    }
    [arr[minIdx], arr[i]] = [arr[i], arr[minIdx]];
  }
  return arr;
}

console.log(
  JSON.stringify(selectionSort([5, 1, 10, 45, 23, 15, 9])) ===
    '[1,5,9,10,15,23,45]'
);

console.log(
  JSON.stringify(selectionSort([10, 9, 8, 23, -19, 5])) ===
    '[-19,5,8,9,10,23]'
);

console.log(
  JSON.stringify(selectionSort([87, -5, -6, 32, 123])) ===
    '[-6,-5,32,87,123]'
);

File: sort/selection.cpp
#include <iostream>


void selectionSort(int arr[], int size) {
    std::cout << "Before: ";
    for (int i = 0; i < size; i++) {
        std::cout << arr[i] << " ";
    }

    int minIdx = 0; 
    
    for (int i = 0; i < size; i++) {
        minIdx = i; 
        for (int j = i + 1; j < size; j++) {
            if (arr[minIdx] > arr[j]) {
                minIdx = j; 
            }
        }
        if (i != minIdx) {
            int temp = arr[i]; 
            arr[i] = arr[minIdx]; 
            arr[minIdx] = temp;
        }
        
    }

}

int main() {
    std::cout << "SELECTION SORT" << std::endl;
    int arr [10] = {0, 1, 9, 8, 7, 6, 5, 4, 3, 2};

    selectionSort(arr, 10);
    
    std::cout << "\nAfter: ";
    for (int i = 0; i < 10; i++) {
        std::cout << arr[i] << " ";
    }
    
    std::cout << std::endl << std::endl << std::endl;
}